src/opt/ReplaceHalfMaskedMemOps.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+  Copyright (c) 2024, Intel Corporation
+
+  SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "ReplaceHalfMaskedMemOps.h"
+
+#include <llvm/IR/IRBuilder.h>
+
+namespace ispc {
+
+bool lIsSecondHalfAllFalse(llvm::Value *mask) {
+    if (auto *CV = llvm::dyn_cast<llvm::ConstantVector>(mask)) {
+        auto N = CV->getType()->getNumElements();
+        for (auto i = N / 2; i < N; i++) {
+            llvm::Constant *E = CV->getAggregateElement(i);
+            if (!E || !llvm::isa<llvm::ConstantInt>(E) || !llvm::cast<llvm::ConstantInt>(E)->isZero()) {
```

Comment:
Is it possible to use `llvm::all_of` in this function?

Reply:
I am not quite sure how to use it for aggregates.